### ISSUE TITLE
During database parameter copying ALTER SYSTEM SET seems to operate using the source database name

### ISSUE THREAD

**Issue by kbarber** [dimitri/pgcopydb#508]
Greetings,

I noticed this since this commit: https://github.com/dimitri/pgcopydb/commit/cc702401d0b8c52a8877376b5a776bb207595407

We seem to get this error message when cloning from a source and target database that have different names:

```
2023-10-24 13:21:05 7 INFO   main.c:163                Running pgcopydb version 0.13.41.ga3a577c from "/root/pgcopydb/pgcopydb"
2023-10-24 13:21:05 7 INFO   cli_common.c:1239         [SOURCE] Copying database from "postgres://postgres@source.host:5432/sourcedb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60"
2023-10-24 13:21:05 7 INFO   cli_common.c:1240         [TARGET] Copying database into "postgres://admin@target.host:5432/targetdb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60"
2023-10-24 13:21:05 7 INFO   snapshot.c:100            Exported snapshot "0000000D-00000123-1" from the source database
2023-10-24 13:21:05 15 INFO   cli_clone_follow.c:530    STEP 1: fetch source database tables, indexes, and sequences
2023-10-24 13:21:05 15 INFO   copydb_schema.c:143       Fetched information for 2 extensions
2023-10-24 13:21:06 15 INFO   copydb_schema.c:282       Splitting source candidate tables larger than 47 MB
2023-10-24 13:21:06 15 INFO   copydb_schema.c:440       Fetched information for 50 tables, with an estimated total of 2990  tuples and 633 MB
2023-10-24 13:21:06 15 INFO   copydb_schema.c:466       Fetched information for 50 indexes
2023-10-24 13:21:06 15 INFO   sequences.c:41            Fetching information for 50 sequences
2023-10-24 13:21:06 15 WARN   pgsql.c:608               Failed to connect to "postgres://admin@target.host:5432/targetdb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60", retrying until the server is ready
2023-10-24 13:21:06 15 INFO   pgsql.c:693               Successfully connected to "postgres://admin@target.host:5432/targetdb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60" after 2 attempts in 58 ms.
2023-10-24 13:21:06 15 INFO   cli_clone_follow.c:547    STEP 2: dump the source database schema (pre/post data)
2023-10-24 13:21:06 15 INFO   pgcmd.c:469                /usr/bin/pg_dump -Fc --snapshot 0000000D-00000123-1 --section pre-data --file /tmp/pgcopydb/schema/pre.dump 'postgres://postgres@source.host:5432/sourcedb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60'
2023-10-24 13:21:06 15 INFO   pgcmd.c:469                /usr/bin/pg_dump -Fc --snapshot 0000000D-00000123-1 --section post-data --file /tmp/pgcopydb/schema/post.dump 'postgres://postgres@source.host:5432/sourcedb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60'
2023-10-24 13:21:07 15 INFO   cli_clone_follow.c:559    STEP 3: restore the pre-data section to the target database
2023-10-24 13:21:07 15 WARN   pgsql.c:608               Failed to connect to "postgres://admin@target.host:5432/targetdb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60", retrying until the server is ready
2023-10-24 13:21:07 15 INFO   pgsql.c:693               Successfully connected to "postgres://admin@target.host:5432/targetdb?keepalives=1&keepalives_idle=10&keepalives_interval=10&keepalives_count=60" after 2 attempts in 54 ms.
2023-10-24 13:21:07 15 INFO   dump_restore.c:299        ALTER DATABASE "sourcedb" SET "search_path" TO '$user', 'public', 'extra';
2023-10-24 13:21:07 15 ERROR  pgsql.c:1965              [TARGET 21102] ERROR:  database "sourcedb" does not exist
2023-10-24 13:21:07 15 ERROR  pgsql.c:1965              [TARGET 21102] ERROR:  database "sourcedb" does not exist
2023-10-24 13:21:07 15 ERROR  pgsql.c:1976              [TARGET 21102] SQL query: ALTER DATABASE "sourcedb" SET "search_path" TO '$user', 'public', 'extra';
2023-10-24 13:21:07 15 ERROR  pgsql.c:1984              [TARGET 21102] SQL params:
2023-10-24 13:21:07 15 ERROR  dump_restore.c:152        Failed to restore the database properties, see above for details
2023-10-24 13:21:07 15 ERROR  cli_clone_follow.c:460    Failed to clone source database, see above for details
2023-10-24 13:21:07 7 ERROR  cli_clone_follow.c:725    clone process 15 has terminated [6]
```

When rolling back to an older version the problem goes away. I'm curious, Is there a way to change this, or is this a bug of some kind? The source and target expected name is specified in the connection string already.

Thanks. I just want to say I enjoy using your tool, great job :-).

ken.

**Comment by VaibhaveS** [comment]
I think we should add a parameter like `--database-properties` in clone and apply it only when the source and target database names are same

**Comment by kbarber** [comment]
I've been testing using this today: https://github.com/dimitri/pgcopydb/pull/509/files I'll run it through our test suite a bit more.

**Comment by dimitri** [comment]
Fixed in #509
